G/gmime/build_tarballs.jl

```diff
@@ -0,0 +1,49 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "gmime"
+version = v"3.2.15"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/jstedfast/gmime/releases/download/$(version)/gmime-$(version).tar.xz", "84cd2a481a27970ec39b5c95f72db026722904a2ccf3fdbd57b280cf2d02b5c4")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/gmime*
+echo '#define ICONV_ISO_INT_FORMAT "iso-%u-%u"' > iconv-detect.h
+echo '#define ICONV_ISO_STR_FORMAT "iso-%u-%s"' >> iconv-detect.h
+echo '#define ICONV_SHIFT_JIS "shift-jis"' >> iconv-detect.h
+echo '#define ICONV_10646 "UCS-4BE"' >> iconv-detect.h
+
+export LDFLAGS="-liconv"
+export CMAKE_EXE_LINKER_FLAGS="-Wl,--copy-dt-needed-entries"
```

Comment:
I'm still not clear why this is necessary, we never had any package needing something like this.

Reply:
There is building error without it
![image](https://github.com/user-attachments/assets/f4cfd5b8-f07e-48f9-8ced-b6527dc31230)

Reply:
I can compile for `x86_64-linux-gnu` without this line.
```suggestion
```

Reply:
As demonstrated at https://github.com/JuliaPackaging/Yggdrasil/pull/9427#issuecomment-2361981292 this flags is not needed at all.